### PR TITLE
Dashboard changes

### PR DESCRIPTION
**Pull request by abdlrhman08**
- [x] Added total spent
- [x] Transactions get listed on the dashboard page

### CODE REVIEW

File: frontend/src/common/LoginController.ts
Comment (Deadreyo) [inline frontend/src/common/LoginController.ts:1]: Guest User logic will fail when the backend is connected. Revert the changes after you finish testing with it.

File: frontend/src/dashboard/DashboardPageController.ts [hunk 1]
```diff
@@ -1,10 +1,66 @@
...
     
-}
\ No newline at end of file
+    this.transactionsTable = <HTMLTableElement>document.getElementById("trans-table")
+    this.amountSpentSpan = document.getElementById("amount");
+    this.createDummyPayments()
+
+    this.userTransactions = this.user.getAllTransactions()
+    const totalSpent =this.userTransactions.reduce((prev, current) => {
+      return prev + current.getAmount()
```
Comment (Deadreyo) [inline frontend/src/dashboard/DashboardPageController.ts:21]: Not all transactions are of type "expense". Check the type of the transaction before adding it to `totalSpent`

**Review by Deadreyo — CHANGES_REQUESTED** [review]

File: frontend/src/common/LoginController.ts
Comment (Deadreyo) [inline frontend/src/common/LoginController.ts:1]: Remove all the guest user logic. Debug with a normal account.

**Review by Deadreyo — CHANGES_REQUESTED** [review]

File: frontend/src/common/LoginController.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export default class LoginController {
     }
 
     checkSavedUser() {
-        const savedUser = localStorage.getItem(this.savedUserKey)
+        const savedUser = localStorage.getItem(this.dbKey)
         if (!savedUser) {
```
Comment (abdlrhman08) [inline frontend/src/common/LoginController.ts:43]: Adding transactions saves the to the db not to the user, so retrieving the user works only if i get it from the db

File: frontend/src/common/Classes/Transaction.ts [hunk 3]
```diff
@@ -34,6 +39,11 @@ export default abstract class Transaction {
     return this.creationDate;
   }
 
+  public setCreationDate(date: number) {
+    this.creationDate = date;
+    this.save();
+  }
```
Comment (Deadreyo) [inline frontend/src/common/Classes/Transaction.ts:45]: Creation date must not be modified, that's why it didn't have a setter. Delete this method and any logic that enables changing it.

File: frontend/public/dashboard.html
Comment (Deadreyo) [inline frontend/public/dashboard.html:1]: Tables do not refresh after changing a transaction's type.

Example steps:
1- Edit a transaction
2- Change type then close
Observation: Transaction didn't move to the correct table
A quick solution could be to refresh all transactions

Comment (Deadreyo) [inline frontend/public/dashboard.html:1]: https://github.com/Deadreyo/software-project/assets/52013393/7838755b-0811-4bc8-8983-fb1e12ec5257

What is "Type"? Is it Periodic vs OneTime, or is it Expense vs Income?
It starts as Periodic but changing the category changes it to Expense vs Income.

**Review by Deadreyo — CHANGES_REQUESTED** [review]

File: frontend/public/dashboard.html
Comment (abdlrhman08) [reply frontend/public/dashboard.html:1]: bro thinks i work a 9 to 5

Comment (Deadreyo) [inline frontend/public/dashboard.html:1]: There should be columns for `date` of OneTime, and `start date` of Periodic. The current Date column reflects `creation date` but it isn't obvious and need to be renamed. Alternatively, you can simply not show the creation date at all.

Comment (Deadreyo) [inline frontend/public/dashboard.html:1]: image

I am trying to edit a Periodic Transaction and periodic fields like `start date`, `interval` and `limit` do not show up to be edited.

Comment (Deadreyo) [inline frontend/public/dashboard.html:1]: image

For a OneTime Transaction, I expect the `date` field to be edited is to be the date of the transaction. I think it is the set to `creation date`, and changing it doesn't save it.

**Review by Deadreyo — CHANGES_REQUESTED** [review]

File: frontend/public/dashboard.html
Comment (abdlrhman08) [reply frontend/public/dashboard.html:1]: I like the second option

**Review by Deadreyo — APPROVED** [review]

**Pull request merged**